File: ChatClientVB/Utils/ViewModelLocator.vb
Imports Unity

Public Class ViewModelLocator
    Private container As UnityContainer

    Public Sub New()
        container = New UnityContainer
        container.RegisterType(Of IChatService, ChatService)()
        container.RegisterType(Of IDialogService, DialogService)()
    End Sub

    Public ReadOnly Property MainVM As MainWindowViewModel
        Get
            Return container.Resolve(Of MainWindowViewModel)()
        End Get
    End Property
End Class

File: ChatClientCS/Utils/ViewModelLocator.cs
using Unity;
using ChatClientCS.Services;
using ChatClientCS.ViewModels;

namespace ChatClientCS.Utils
{
    public class ViewModelLocator
    {
        private UnityContainer container;

        public ViewModelLocator()
        {
            container = new UnityContainer();
            container.RegisterType<IChatService, ChatService>();
            container.RegisterType<IDialogService, DialogService>();
        }

        public MainWindowViewModel MainVM
        {
            get { return container.Resolve<MainWindowViewModel>(); }
        }
    }
}
